backend/de.metas.manufacturing/src/main/java/org/eevolution/productioncandidate/service/CreateOrderCandidateCommand.java

```diff
@@ -118,6 +122,17 @@ public I_PP_Order_Candidate execute()
 		return ppOrderCandidateRecord;
 	}
 
+	private int calculatePPOrderCandidateSeqNo(final I_PP_Order_Candidate ppOrderCandidateRecord)
+	{
+		final int lastSeqNo = Services.get(IQueryBL.class)
+				.createQueryBuilder(I_PP_Order_Candidate.class, ppOrderCandidateRecord)
+				.addEqualsFilter(I_PP_Order_Candidate.COLUMNNAME_M_Product_ID, ppOrderCandidateRecord.getM_Product_ID())
```

Comment:
i can't see where the requirement regarding "same product" is coming from. to me it seems useless at best and might be detrimental if the user wants to run different products through the same machine (which many users don't but some might)
related idea: why don't we 
- make `OrderCandidateSeqNo` an SQL-column with the value being `PP_Order_Candidate_ID * 10`
- and sort the processable candidates by `PP_Order_Candidate_ID` 

please **clarify this question** and **write a functional concept** before requesting further code-reviews from me